lib/game_team.rb

```diff
@@ -37,6 +41,58 @@ def self.winningest_coach
   def self.worst_coach
   end
 
+  def self.game_team_shots_goals_count(arr_games)
+    season = arr_games.first.game_id
+    self.find_by(season)
+  end
+
+  def self.get_goal_shots_by_game_team(game_teams)
+    #passes in an array of games
+    results = {}
+    #each game, iterating through and adding the team_id as the key and
+    #getting the shots and goals count
+    game_teams.each do |game_team|
+      results[game_team.team_id] ||= {"shots"=>0,"goals"=>0}
```

Comment:
does `||=` assign `{"shots"=>0,"goals"=>0}` if the array key is nil?

Reply:
Kind of. What it's doing is saying if the values don't exist for the key `results[game_team.team_id]` then assign the hash `{"shots"=>0,"goals"=>0}` as it's values creating a subhash or hash within a hash aka hashception

Reply:
memoization in the house!